### PR TITLE
Cast array params to np array before inserting

### PR DESCRIPTION
**Pull request by jenshnielsen**
Otherwise this will not work correctly when inserting as arrays

### CODE REVIEW

File: qcodes/dataset/measurements.py [hunk 1]
```diff
@@ -174,9 +174,10 @@ def add_result(self, *res_tuple: res_type) -> None:
                 inserting_as_arrays = True
                 inserting_this_as_array = True
             if any(isinstance(value, typ) for typ in array_like_types):
-
                 value = cast(np.ndarray, partial_result[1])
                 value = np.atleast_1d(value)
+                # we want to insert the casted version
```
Comment (astafan8) [inline qcodes/dataset/measurements.py]: it's not only casted but also converted, right? for example, if `partial_result[1]` is a tuple, the tuple gets also converted to numpy array by `np.atleast_1d`? or does `cast(np.ndarray, ...` do that already?

Comment (jenshnielsen) [reply qcodes/dataset/measurements.py]: Converting from one type to another is a cast. However the function `cast` is indeed badly named and does not perform any real cast but purely changes the nominal type. The real `cast` is indeed performed by `atleast1d` as you say

Comment (astafan8) [reply qcodes/dataset/measurements.py]: thanks, good to know. i got pythonized too much))

File: qcodes/tests/dataset/test_measurement_context_manager.py [hunk 2]
```diff
@@ -800,158 +800,151 @@ def test_datasaver_foul_input():
...
 
 @settings(max_examples=5, deadline=None)
 @given(N=hst.integers(min_value=5, max_value=500),
-       M=hst.integers(min_value=4, max_value=250))
+       M=hst.integers(min_value=4, max_value=250),
+       seed=hst.integers(min_value=0, max_value=np.iinfo(np.uint32).max))
 @pytest.mark.usefixtures("experiment")
-def test_datasaver_array_parameters(SpectrumAnalyzer, DAC, N, M):
-    spectrum = SpectrumAnalyzer.spectrum
+@pytest.mark.parametrize("param_type", ['array', 'tuple', 'list'])
```
[the remark this answers is not in the document]
Comment (astafan8) [reply qcodes/tests/dataset/test_measurement_context_manager.py]: thanks!

**Comment by jenshnielsen** [comment]
I believe that coverage drops because https://github.com/QCoDeS/Qcodes/blob/master/qcodes/dataset/measurements.py#L264 is no longer hit (the conversion is now taken care of by atleast_1d)

**Pull request merged**